components/Common/NavBar.tsx

```diff
@@ -59,6 +66,10 @@ const NavBar: FC = () => {
     if (_.isEmpty(blogs)) {
       dispatch(getAllBlogs());
     }
+
+    if (_.isEmpty(sponsors)) {
+      dispatch(getAllSponsors());
+    }
```

Comment:
> For sponsors I switched to useSponsors in hooks. That is what u meant right?

yeah, but we should consume the data without dispatch / redux stuff. 

heres (not the best) an example of how we should use the hooks https://github.com/utmmcss/deerhacks/commit/d566e4e657702166fd245ba0f47a6da177ca3721#diff-aa98fd0757d0e1741503c50cfafb7726939d19819638bbe8e030a27adfec34a3R20

but instead we should return something like 

```typescript
if loading return <some loading component>
if error return <some error thing>

// this is where we know the data is valid & is there
return ...
```
![image](https://github.com/utmmcss/mcss-website-frontend/assets/70813519/53e02c6f-3d09-425b-ba97-9fa752ff6f80)

Reply:
heres the official docs example https://tanstack.com/query/v4/docs/react/guides/queries#:~:text=tsx-,function%20Todos()%20%7B,%7D,-If%20booleans%20aren%27t